business/istio_status.go

```diff
@@ -100,46 +106,28 @@ func (iss *IstioStatusService) getIstioComponentStatus(ctx context.Context, clus
 			IsCore:    true,
 		})
 	}
-	deploymentStatus = deploymentStatus.Merge(istiodStatus)
-	deploymentStatus = deploymentStatus.Merge(iss.getGatewaysStatus(ctx, cluster))
 
-	return deploymentStatus, nil
-}
+	// Autodiscover gateways.
+	workloads, err := iss.workloads.GetAllWorkloads(ctx, cluster)
```

Comment:
I'm not sure is this could be more efficient but just asking: 
All the workloads are fetched just to see if they are gateways, if the gateways are identified by labels, it would be better to create a GetAllGateways method that fetches all the gateway labels?